dot/network/host.go

```diff
@@ -187,7 +187,7 @@ func newHost(ctx context.Context, cfg *Config) (*host, error) {
 	}
 
 	// create libp2p host instance
-	h, err := libp2p.New(ctx, opts...)
+	h, err := libp2p.New(opts...)
```

Comment:
Don't we need to handle the context cancellation explicitly now and call `h.Close()`?

Reply:
We are handling it in our close function at line 239 in dot/host.go